google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/Exceptions.java

```diff
@@ -71,13 +71,22 @@ public long getActualOffset() {
     }
   }
 
-  /** Stream has already been finalized. */
+  /**
+   * Stream created by CreateWriteStream has already been finalized. You need to recreate a stream
+   * using CreateWriteStream.
+   */
   public static final class StreamFinalizedException extends StorageException {
     protected StreamFinalizedException(Status grpcStatus, String name) {
       super(grpcStatus, name, null, null, ImmutableMap.of());
     }
   }
 
+  /**
+   * This writer has been closed either by user or by some non retry errors we encounter. You need
```

Comment:
This writer instance has either been closed by the user explicitly, or has encountered non-retriable errors.

To continue to write to the same stream, you will need to create a new writer instance.